src/pages/inventory/inventory.ts

```diff
@@ -40,12 +54,31 @@ export class InventoryPage {
   }
 
   filterItems() {
-    this.inventoryData.filterItems(this.selectedCategoryIDs, this.segment).subscribe(
-      items => this.filteredItems = items,
+    this.filterModels();
+
+    if (Math.sign(this.selectedBrandID) < 0) {
+      this.selectedModelID = -1;
+    }
+
+    this.inventoryData.filterItems(
+      this.selectedBrandID,
+      this.selectedModelID,
+      this.selectedCategoryID,
+      this.segment
+    ).subscribe(
+      items => {
+        this.filteredItems = items.results;
+      },
       err => this.stockpileData.showToast(err.message)
     );
   }
 
+  filterModels() {
```

Comment:
It feels a little weird to have such simple operations moved to the back end, but you are right that it would make things more consistent.